controllers/dspipeline_controller.go

```diff
@@ -294,6 +305,17 @@ func (r *DSPAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 			return ctrl.Result{}, err
 		}
 
+		if dspa.Spec.APIServer.PipelineStorage == "kubernetes" && r.DeployWebhook {
+			err = r.ReconcileWebhook(ctx, dspa, params)
+			if err != nil {
+				r.setStatusAsNotReady(config.APIServerReady, err, dspaStatus.SetApiServerStatus)
```

Comment:
Could we have separate conditions for the webhook status in the DSPA like we do for `DatabaseAvailable and `ObjectStoreAvailable`?